### PR TITLE
Feature/companion

### PR DESCRIPTION
**Pull request by atharvagupta2003**
-basic UI change
-minor bugs solved

### CODE REVIEW

**Review by aniketgupta17 — APPROVED** [review]
seems fine

File: source/core/src/main/com/csse3200/game/components/companion/CompanionActions.java
Comment (Foref) [inline source/core/src/main/com/csse3200/game/components/companion/CompanionActions.java:1]: Check indentation of curly brackets

File: source/core/assets/levels/flora_haven/main_area/entities/laboratoryhaven.json
Comment (Foref) [inline source/core/assets/levels/flora_haven/main_area/entities/laboratoryhaven.json:1]: check string "class": "pathname"

File: source/core/assets/levels/flora_haven/main_area/entities/companion.json
Comment (Foref) [inline source/core/assets/levels/flora_haven/main_area/entities/companion.json:1]: check string "class": "pathname"

File: source/core/assets/levels/cryoheim/main_area/entities/Laboratorycryoheim.json
Comment (Foref) [inline source/core/assets/levels/cryoheim/main_area/entities/Laboratorycryoheim.json:1]: check string "class": "pathname"

**Comment by Foref** [comment]
Conflicts and json file string error not resolved yet, please change them before merge!!!!!!

**Review by Foref — CHANGES_REQUESTED** [review]
Please change the errors listed + merge conflicts

File: source/core/assets/levels/flora_haven/main_area/entities/companion.json
Comment (Foref) [reply source/core/assets/levels/flora_haven/main_area/entities/companion.json:1]: the path needs to be a string too,
"com.csse3200.game.entities.configs.CompanionConfig"

File: source/core/assets/levels/cryoheim/main_area/entities/Laboratorycryoheim.json
Comment (Foref) [reply source/core/assets/levels/cryoheim/main_area/entities/Laboratorycryoheim.json:1]: the path needs to be a string too,
example here,

 "com.csse3200.game.entities.configs.CompanionConfig"

File: source/core/assets/levels/flora_haven/main_area/entities/laboratoryhaven.json
Comment (Foref) [reply source/core/assets/levels/flora_haven/main_area/entities/laboratoryhaven.json:1]: the path needs to be a string too,
example here,

"com.csse3200.game.entities.configs.CompanionConfig"

File: source/core/src/main/com/csse3200/game/components/companion/CompanionActions.java
Comment (Foref) [reply source/core/src/main/com/csse3200/game/components/companion/CompanionActions.java:1]: line 328

Comment (Foref) [reply source/core/src/main/com/csse3200/game/components/companion/CompanionActions.java:1]: line 328 curly bracket indentation

**Comment by Foref** [comment]
CompanionActions line328 + merge conflicts again

**Pull request closed without merging**